Lib/test/test_builtin.py

```diff
@@ -629,8 +629,9 @@ def __dir__(self):
 
     def test___ne__(self):
         self.assertFalse(None.__ne__(None))
-        self.assertTrue(None.__ne__(0))
-        self.assertTrue(None.__ne__("abc"))
+        with self.assertWarnsRegex(DeprecationWarning, "NotImplemented should not be used in a boolean context"):
+            self.assertTrue(None.__ne__(0))
+            self.assertTrue(None.__ne__("abc"))
```

Comment:
```suggestion
        msg = "NotImplemented should not be used in a boolean context"
        with self.assertWarnsRegex(DeprecationWarning, msg):
            self.assertTrue(None.__ne__(0))
        with self.assertWarnsRegex(DeprecationWarning, msg):
            self.assertTrue(None.__ne__("abc"))
```

Reply:
Why not test against the real result?
```py
self.assertIs(None.__ne__(0), NotImplemented)
```
Today's DeprecationWarning is tomorrow's error.

Reply:
> Why not test against the real result?

I wondered about that, but decided maybe that was a bad idea, since the test is a regression test for https://github.com/python/cpython/issues/112125, and so the important thing is that the expression is truthy, not that the result is `NotImplemented`. But I don't feel strongly.

Reply:
```py
        msg = "NotImplemented should not be used in a boolean context"
        with self.assertWarnsRegex(DeprecationWarning, msg):
            self.assertTrue(None.__ne__(0))
```
is a more complicated way to test that the result is `NotImplemented`.

Reply:
I suppose so.

Reply:
> Why not test against the real result?
> 
> ```python
> self.assertIs(None.__ne__(0), NotImplemented)
> ```
> 
> Today's DeprecationWarning is tomorrow's error.

Indeed, for me this looks more suitable.